src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -180,58 +182,77 @@ public int Seek(Vector<byte> byte0Vector)
 
             var block = _block;
             var index = _index;
-            var array = block.Array;
+            var following = block.End - index;
+            byte[] array;
+            var byte0 = byte0Vector[0];
+
             while (true)
             {
-                while (block.End == index)
+                while (following == 0)
                 {
-                    if (block.Next == null)
+                    var newBlock = block.Next;
+                    if (newBlock == null)
                     {
                         _block = block;
                         _index = index;
                         return -1;
                     }
-                    block = block.Next;
-                    index = block.Start;
-                    array = block.Array;
+                    index = newBlock.Start;
+                    following = newBlock.End - index;
+                    block = newBlock;
                 }
-                while (block.End != index)
+                array = block.Array;
+                while (following > 0)
                 {
-                    var following = block.End - index;
-                    if (following >= Vector<byte>.Count)
+#if !DEBUG 
+// Need unit tests to test Vector path
+                    // Check will be Jitted away https://github.com/dotnet/coreclr/issues/1079
+                    if (Vector.IsHardwareAccelerated)
                     {
-                        var data = new Vector<byte>(array, index);
-                        var byte0Equals = Vector.Equals(data, byte0Vector);
-
-                        if (byte0Equals.Equals(Vector<byte>.Zero))
+#endif
+                        if (following >= _vectorSpan)
```

Comment:
@benaadams After initialize is done, later compiled functions will have hardcoded 10h.
Because you already check on IsHardwareAccelerated you can safely remove _vectorSpan and replace it with Vector<byte>.Count